opslevel/resource_opslevel_alias.go

```diff
@@ -0,0 +1,246 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &AliasResource{}
+
+func NewAliasResource() resource.Resource {
+	return &AliasResource{}
+}
+
+// AliasResource defines the resource implementation for managing aliases.
+type AliasResource struct {
+	CommonResourceClient
+}
+
+func (r *AliasResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_alias"
+}
+
+type AliasResourceModel struct {
+	ResourceType       types.String `tfsdk:"resource_type"`
+	ResourceIdentifier types.String `tfsdk:"resource_identifier"`
+	Aliases            types.Set    `tfsdk:"aliases"`
+
+	Id types.String `tfsdk:"id"`
+}
+
+func (s AliasResourceModel) GetResource(d *diag.Diagnostics, client *opslevel.Client) opslevel.AliasableResourceInterface {
+	resourceType := opslevel.AliasOwnerTypeEnum(s.ResourceType.ValueString())
+	resourceIdentifier := s.ResourceIdentifier.ValueString()
+	output, err := client.GetAliasableResource(resourceType, resourceIdentifier)
+	if err != nil {
+		d.AddError(
+			"opslevel client error",
+			fmt.Sprintf("Failed to find aliasable resource, %s", err),
+		)
+	}
+	return output
+}
+
+func (s AliasResourceModel) GetAliases(ctx context.Context, d *diag.Diagnostics) []string {
+	var output []string
+	if !s.Aliases.IsNull() {
```

Comment:
```suggestion
	if !s.Aliases.IsNull() && !s.Aliases.IsUnknown() {
```
Not sure if this is actually needed but it might avoid a future bug

Reply:
Tested manually with

```
resource "random_pet" "example" {
}

resource "opslevel_alias" "system" {
    resource_type       = "system"
    resource_identifier = "internal_services"

    aliases = [random_pet.example.id]
}
```

resulting in

```
  # opslevel_alias.system will be created
  + resource "opslevel_alias" "system" {
      + aliases             = [
          + (known after apply),
        ]
      + id                  = (known after apply)
      + resource_identifier = "internal_services"
      + resource_type       = "system"
    }
```

It didn't need this so skipping.